src/datasets/containers.jl

```diff
@@ -78,8 +79,8 @@ function LearnBase.nobs(dataset::TableDataset{T}) where {T}
     end
 end
 
-LearnBase.getobs(dataset::TableDataset{<:DataFrame}, idx) = dataset.table[idx, :]
+LearnBase.getobs(dataset::TableDataset{<:DataFrame}, idx) = [data for data in dataset.table[idx, :]]
```

Comment:
Yeah for `DataFrame` this would make more sense. But is it fine to keep this return behaviour for `getobs(dataset::TableDataset{<:CSV.File}, idx)` and the general `getobs(dataset::TableDataset{T}, idx)` methods considering that the types could be immutable, and they may not not have `setindex` defined (as in the case of `CSV.Row`) which might make transformations difficult if we use a copy of the input directly?

Reply:
I think any transforms shouldn't be calling `setindex!` anyhow without a defensive copy first, so we should stick with `dataset.table[idx, :]` everywhere.

Reply:
Agreeing with Brian that the transforms shouldn't mutate the arguments by default; we have `encode!`/`run!` for the inplace versions.

Reply:
+1 for the existing suggestions